Adding RunOn support and other issue fixes.

vrdmr opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Adding runOn support for Scheduled Job and Webhook cmdlets; Also adding alias for Modules in New/Set Modules. The changes were required for couple of issues raised earlier.
<!--
Please add an informative description that covers the changes made by the pull request.

If applicable, reference the bug/issue that this pull request fixes here.
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

msftclas commented:
@vrdmr,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

azuresdkci commented:
Can one of the admins verify this patch?

cormacpayne commented:
@azuresdkci add to whitelist

markcowl commented:
@vrdmr There has been no activity on this PR in 10 days, we will have to close it this week if you do not address review feedback

vrdmr commented:
@cormacpayne @markcowl 

- Do you plan to add test coverage for the cmdlets that are now using RunOn?
Updated the existing tests. 

- Does this PR resolve issues #3082 and #3197 ?
Yes. #3775, #3197 and #3082.

- Can you update the change log with a snippet about the changes you are making in this PR?
Updated the ChangeLog with a small description of what has been updated.

cormacpayne commented on src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationModule.cs:
```diff
@@ -39,6 +39,7 @@ public class NewAzureAutomationModule : AzureAutomationBaseCmdlet
         /// </summary>
         [Parameter(Position = 3, Mandatory = true, ValueFromPipelineByPropertyName = true,
             HelpMessage = "The url to a module zip package.")]
+        [Alias("ContentLinkUri")]
```
@vrdmr I think it would be better if we changed the name of this parameter to `ContentLinkUri` and added an alias to `ContentLink`. That way, when users tab through the parameters, `ContentLinkUri` will show up and will be consistent with what is shown when tabbing through the parameters of `Set-AzureRmAutomationModule`.

vrdmr replied:
I agree. I'll confirm with @safeermohammed and make the change asap.

safeermohammed replied:
Does changing he parameter name will be breaking change, i.e., does user already using the cmdlet need to do any thing? If this is not the case, fine to do as suggested.

cormacpayne replied:
@vrdmr @safeermohammed if we swap the parameter name and its alias, it won't be a breaking change. Existing scripts will be able to continue to use `ContentLink`; the only difference is when people tab through the possible parameters (as well as look at the docs for the cmdlet), they will see `ContentLinkUri`, which I think is a better indication of what this parameter is. The use of `ContentLink` is not lost in this swap.

safeermohammed replied:
@cormacpayne Thank you for confirmation, we will swap it

vrdmr commented:
@cormacpayne - No particular reason for not passing any specific values for runOn. It is currently not a high priority scenario and we are in process of adding more functionality in the backend for it.

vrdmr commented:
@cormacpayne @markcowl Ping!

cormacpayne commented on src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationModule.cs:
```diff
@@ -39,8 +39,9 @@ public class NewAzureAutomationModule : AzureAutomationBaseCmdlet
+        [Alias("ContentLink")]
         [ValidateNotNullOrEmpty]
-        public Uri ContentLink { get; set; }
+        public Uri ContentLinkUri { get; set; }
```
@vrdmr you are still referencing `ContentLink` below instead of `ContentLinkUri`, which is causing compilation errors

vrdmr replied:
Fixed.

markcowl commented:
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1822/

@vrdmr Filed issue #4422 for adding additional runOn tests

vrdmr commented:
Thanks @cormacpayne! :)
